### DISCUSSION TITLE
error returned from database: relation "proof_generation_details" does not exist

### DISCUSSION THREAD

**Discussion by Apollaman** [zkSync-Community-Hub/zksync-developers#759] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I received error while trying to upgrade image from 24.0.0 to 25.0.0 ? How to fix it?
error: while executing migration 20240619143458: error returned from database: relation "proof_generation_details" does not exist

 image: matterlabs/external-node:v25.0.0

**Comment by alhenaai** [comment]
It seems like you are encountering a database migration issue when upgrading to ZKsync Era node version 25.0.0. The error message indicates that the "proof_generation_details" relation does not exist in your database, which is likely required for the new version.

Here are a few steps you can take to resolve this issue:

1. **Ensure Database Schema is Updated**: Make sure that all previous migrations are applied correctly before updating to v25.0.0. It's possible that a previous migration that creates the "proof_generation_details" table or relation was not executed or completed successfully.

2. **Check Migration Scripts**: Review the migration scripts included with the v25.0.0 update. There should be a script that creates or modifies the "proof_generation_details" relation. Ensure this script is present and correct in the sequence of migrations.

3. **Reapply Migrations**: If possible, try to reapply the migrations to ensure all database changes are correctly implemented. This might involve rolling back to a previous stable state and reapplying the updates.

4. **Consult the Documentation**: Check the official ZKsync documentation or release notes for any specific instructions or prerequisites required for upgrading to version 25.0.0. Sometimes, specific steps or additional configurations are needed that are not part of the automatic update process.

If these steps do not resolve the issue, I recommend reaching out for more specific support or consulting the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth assistance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by perekopskiy** [comment, marked as answer]
Hello @Apollaman. Can you please check what this query returns for your DB:
```
SELECT * FROM _sqlx_migrations
WHERE version = 20230810090211
```
Do you remember doing any manual actions with your DB, specifically deleting `proof_generation_details` table?
Anyway, to fix this issue please run these queries
```
CREATE TABLE IF NOT EXISTS proof_generation_details
(
    l1_batch_number         BIGINT PRIMARY KEY REFERENCES l1_batches (number) ON DELETE CASCADE,
    status                  TEXT      NOT NULL,
    proof_gen_data_blob_url TEXT      NOT NULL,
    proof_blob_url          TEXT,
    created_at              TIMESTAMP NOT NULL,
    updated_at              TIMESTAMP NOT NULL,
    prover_taken_at         TIMESTAMP
);


CREATE INDEX IF NOT EXISTS idx_proof_generation_details_status_prover_taken_at
    ON proof_generation_details (prover_taken_at)
    WHERE status = 'picked_by_prover';
```
Then launch 25.0.0 and it should work
